### PR TITLE
Feature/print context

### PR DESCRIPTION
**Pull request by hoesler**
Adds the possibility to give a DBItest_context a name which will be used test messages.

Example:
```R
DBItest::make_context(..., name = NULL) # default
# BItest: Getting started: ....S

DBItest::make_context(..., name = "read_only")
# BItest[read_only]: Getting started: ....S
```

### CODE REVIEW

File: R/run.R [hunk 1]
```diff
@@ -1,12 +1,15 @@
-run_tests <- function(tests, skip, test_suite) {
-  context(paste("DBItest:", test_suite))
+run_tests <- function(tests, skip, test_suite, ctx_name) {
+  test_context <- paste0(
+    "DBItest", ifelse(is.null(ctx_name), "", paste0("[", ctx_name, "]")),
```
Comment (krlmlr) [inline R/run.R]: ifelse() -> if (!is.null(...)) ...

File: R/test_getting_started.R [hunk 2]
```diff
@@ -24,7 +24,8 @@ test_getting_started <- function(skip = NULL, ctx = get_default_context()) {
     #' \code{make_context()} must be called before calling any of the
     #' \code{test_} functions in this package}
     has_context = function() {
-      expect_is(ctx, "DBItest_context")
+      object <- ctx # object duplication fixes lazyeval error
```
Comment (krlmlr) [inline R/test_getting_started.R]: In what context does the error occur, what is the message? Is it perhaps enough to call `expect_is( (ctx), ...)`?

Comment (hoesler) [inline R/test_getting_started.R]: The error message was
```
1. Error: DBItest[Derby]: has_context ----------------------------------------------------------------------------------
Promise has already been forced
1: test_fun()
2: expect_is(ctx, "DBItest_context")
3: make_label(object, label)
4: label %||% label(object)
5: label(object)
6: lazyeval::lazy(obj)
```

``` (ctx, ...) ``` seems to work. Can give me a hint why?

Comment (krlmlr) [inline R/test_getting_started.R]: An omission in lazyeval, I guess. The expression `(ctx)` lives in the caller's frame, whereas `ctx` lives somewhere else. See https://github.com/hadley/testthat/pull/408 for more context. I'd be glad to see this resolved, but currently it's just a minor corner case that is easy to fix.

**Comment by krlmlr** [comment]
Awesome, thanks!

**Pull request merged**